python/tests/test_aac/plugins/no_ext_for_final/test_no_ext_for_final.py

```diff
@@ -1,38 +1,55 @@
 from unittest import TestCase
-from typing import Tuple
-from click.testing import CliRunner
-from aac.execute.command_line import cli, initialize_cli
-from aac.execute.aac_execution_result import ExecutionStatus
 from aac.context.language_context import LanguageContext
-
-
-from aac.plugins.no_ext_for_final.no_ext_for_final_impl import plugin_name
-
-
 from aac.plugins.no_ext_for_final.no_ext_for_final_impl import no_extension_for_final
 
 
 class TestNoExtForFinal(TestCase):
-    def test_no_extension_for_final(self):
+
+    #  Nominal (expected pass) test with a single valid Schema definition
+    def test_no_extension_for_final_nominal(self):
         context = LanguageContext()
         schema_definition = context.get_definitions_by_name("aac.lang.Schema")
         if len(schema_definition) != 1:
             self.fail("Expected to find one and only one Schema definition")
         schema_definition = schema_definition[0]
+
         definitions = context.parse_and_load(root_schema_with_good_ext)
         result = no_extension_for_final(definitions[0].instance, definitions[0], schema_definition.instance)
         self.assertTrue(result.is_success())
         result = no_extension_for_final(definitions[1].instance, definitions[1], schema_definition.instance)
         self.assertTrue(result.is_success())
         context.remove_definitions(definitions)
 
-        definitions = context.parse_and_load(root_schema_with_bad_ext)
+
+    #  Off-nominal (expected fail) tests
+    def test_no_extension_for_final_offnominal(self):
```

Comment:
I think another valid off nominal case would be trying to extend something that is not defined. So, like you typo the model name you're trying to extend for instance.

Reply:
Yes! I had that one once but lost it when I did a commit in gitpod but not the push. Added back in. Good catch